### ISSUE TITLE
[FR] Bidirectional Links Like Obsidian

### ISSUE THREAD

**Issue by nimalu001** [AppFlowy-IO/appflowy#27]
- navigate from and to different documents through links
- maybe use the same syntax as Obsidian so one could import/export from/to Obsidian
- a graph which visualizes the connections between the linked documents

[Links in Obsidian](https://help.obsidian.md/How+to/Internal+link)

**Comment by Lloyd-Jackman-UKPL** [comment]
I'd absolutely love the graph functionality (am a big fan of Obsidian) but would be keen on even seeing it go one step further by defining linking documents by relationship types that could then be filtered in a graph view.

**Comment by annieappflowy** [comment]
Related: https://github.com/AppFlowy-IO/AppFlowy/issues/1813

**Comment by annieappflowy** [comment]
Related: https://github.com/AppFlowy-IO/AppFlowy/issues/1591

**Comment by skewballfox** [comment]
not either of the above posters but I can answer both as someone who both regularly uses obsidians and has brushed up against its limitations

> What pain points do you feel when using Obsidian's related features to navigate across pages? 
mainly the inability to define aliases to headers sections. This comes up when doing things like defining a glossary, or when you want to have notes on a more abstract concept with multiple components (say like push and pop as related to stacks and queues), it comes up regularly when taking notes related to math or the programming

> can you elaborate on defining linking documents by relationship types? What goals are you trying to achieve with such features? Do relationship types refer to edges in the graph that connect to documents?

again, not the OP so I can't answer this for him, but for me this comes up when doing things like using obsidian to map out an ontology. when I was taking notes for algorithms, I would have greatly appreciated a qualifier for links such as is_a (0-1 knapsack is_a greedy algorithm), as it currently exist in obsidian, all links are equal(ly nondescript). While this is useful, it becomes less so when there is more than one type of relationship between the linked notes/concepts and current.

right now, I have a ton of notes linking to the definition of an invertible matrix, most of these are statements where some result is an invertible matrix, some where it's required of an input, one is the theorem associated with it(which is more a listing of properties), one is more eli5 notes on how to find it. There's currently no way in obsidian to make these relationship types distinct, other than explicitly adopting some structure to how your notes are composed.

**Comment by almereyda** [comment]
Duplicates:

- #1813
- #2214
